server/src/main/java/org/elasticsearch/plugins/ActionPlugin.java

```diff
@@ -61,6 +62,13 @@ default Collection<ActionFilter> getActionFilters() {
         return Collections.emptyList();
     }
 
+    /**
+     * Action filters applying to a single action added by this plugin.
+     */
+    default Collection<MappedActionFilter> getMappedActionFilters() {
```

Comment:
A Q: plugins should migrate to this method to return a MappedActionFilter.
Would it be possible then to remove the ability to return a MappedActionFilter from `getActionFilters()`? I think it will be OK for modules and for serverless, but what about 3rd party? We do not guarantee any bwc there, correct? (They work only with the ES version they were compiled with, to the patch?)

Reply:
Plugins have zero backcompat guarantees (at least these classic plugins). After converting serverless to use MappedActionFilter I will followup with making MappedActionFilter no longer inherit from ActionFilter, so it will not be possible to return via `getActionFilters()`.